### ISSUE TITLE
spurious unused warning of local given for CanEqual with -Wunused:locals and -language:strictEquality

### ISSUE THREAD

**Issue by felher** [lampepfl/dotty#17762]
3.3 is awesome, thank you for bringing warnings back and all your hard work! :heart:
There seems to be a small interaction problem with what is regarded unused and `-language:strictEquality` though:

## Compiler version

3.3.0

## Minimized code

CompilerOptions:
* -language:strictEquality
* -Wunused:locals

```Scala
class SomeType

def testIt(st1: SomeType, st2: SomeType): Boolean =
  given CanEqual[SomeType, SomeType] = CanEqual.derived
  st1 == st2
```

## Output

A warning that the `given` definition is unused - or a compiler error that `SomeType` can not be compare with itself when removed

## Expectation

It should not warn since the `given CanEqual[...]` is required  by the compiler under `-language:strictEquality`

## Scastie
https://scastie.scala-lang.org/i6Osmdc1SSK9FQlN3a65dA

**Comment by som-snytt** [comment]
Implicit behaviors are difficult to emulate. I wonder if the the mechanism could delete the element and type check just that revised compilation unit before issuing a warning. That would insulate the lint from "unhandled" behaviors and also from future changes it can't anticipate.

**Comment by aflierl** [comment]
I would like to remark that the same happens when using `-Wunused:imports` if the `given` is imported from somewhere else (like another package), e.g.:

```scala
object OneModule:
  import AnotherModule.given

  def meep =  Meep("meep") == Meep("moop")

final case class Meep(moop: String)

object AnotherModule:
  given CanEqual[Meep, Meep] = CanEqual.derived
```

Or is this / should this be another bug?

**Comment by joan38** [comment]
Hey! Getting hit by the same issue here.

**Comment by joan38** [comment]
Nice! Is this targeted to be released in 3.3.2?

**Comment by Kordyjan** [comment]
> Is this targeted to be released in 3.3.2?

Yes.
